src/njoy21/input/WIMSR/Card3/Rdfid.hpp

```diff
@@ -5,7 +5,7 @@ struct Rdfid{
     return "The rdfid parameter is the identification number for the\n"
            "resonance data in the output of the WIMS library.";
   }
-  static bool verify( const Value_t rdfid ){
-    return ( rdfid > 0.0 and rdfid < 10000.0 );
+  static bool verify( Value_t rdfid, const int mat ){
+    return rdfid == double(mat);
```

Comment:
I don't believe these have to be equal. That being said, there isn't much information about what is an invalid `rdfid`. I think your previous verify was better.